src/model.ts

```diff
@@ -367,6 +367,15 @@ export class SpecQueryModel {
     return undefined;
   }
 
+  public getEncodingQueryIndexByChannel(channel: Channel) {
```

Comment:
This is O(n) and you're calling it multiple times.  So this is very inefficient.  Remove this method and just create an index by reducing `specM.spec.encodings()`.